### PR TITLE
allow authentication with github token

### LINKED ISSUE #8: clone private repositories
In #6 you wrote...

> No, I'm using anonymous access.  This means, of course, that ghcloneall is only suitable for cloning open source repositories that are publicly available.  

1. oh no!
2. how does this go together with the optional argument "--include-private     include private repositories (default)"? (see readme)

### PR DESCRIPTION
**Pull request by cgroschupp**
fix #8

### CODE REVIEW

**Comment by cgroschupp** [comment]
@mgedmin What do you think?

**Comment by mgedmin** [comment]
(Sorry, I'm currently on vacation and trying to forget that computers exist.  I'll take a look later!)

**Review by mgedmin — APPROVED** [review]
I like this a lot!

Do you think you could add the missing documentation bits (short changelog notice, synopsis section in the README), or would you prefer that I did them myself?

File: ghcloneall.py [hunk 1]
```diff
@@ -776,6 +782,9 @@ def _main():
     parser.add_argument(
         '--user',
         help='specify the GitHub user')
+    parser.add_argument(
+        '--github-token',
+        help='specify the GitHub token')
```
Comment (mgedmin) [inline ghcloneall.py:788]: It also would be nice if ghcloneall could load the token from the config file (this needs like three additional lines of code a page or two down from here).

For completeness maybe `ghcloneall --init --github-token XYZZY` should also _write_ the token to the newly created config file (another two lines of code a bit further down).

**Comment by cgroschupp** [comment]
@mgedmin sorry for the delay

**Review by mgedmin — APPROVED** [review]
Thank you!

I'm going to merge this, but I'd like to have a bit more documentation.  Perhaps a new section in the readme, "Dealing with private repositories"?  It should mention where on GitHub you can get a token, and which scopes are needed.

**Pull request merged**